Adds the release announcement post for v7 of the CLI for Microsoft 365 (#1572)
pnp/blog: content/cli-for-microsoft-365/cli-for-microsoft-365-v7-0/index.md

A part of working with lists is the ability to add attachments to list items, making it easy to keep all relevant information in one place. In this release we bring support for managing list attachments using the CLI for Microsoft 365. 

To add a new attachment to a list item from a local file, run:

```sh
m365 spo listitem attachment add --webUrl https://contoso.sharepoint.com/sites/project-x --listTitle "DemoList" --listItemId 147 --filePath "C:/Reports/File1.jpg"
```

To get information about a list item attachment, run:

```sh
m365 spo listitem attachment get --webUrl https://contoso.sharepoint.com/sites/project-x --listTitle "Demo List" --listItemId 147 --fileName "File1.jpg"
```

To update an attachment, run:

```sh
m365 spo listitem attachment set --webUrl https://contoso.sharepoint.com/sites/project-x --listTitle "Demo List" --listItemId 147 --fileName "File1.jpg" --filePath "C:/Reports/File2.jpg"
```

To remove an attachment from a list item, run:

```sh
m365 spo listitem attachment remove --webUrl https://contoso.sharepoint.com/sites/project-x --listTitle "Demo List" --listItemId 147 --fileName "File1.jpg"
```

> To learn more about managing list item attachments, see the [list item attachment commands](https://pnp.github.io/cli-microsoft365/cmd/spo/listitem/listitem-attachment-list) in our docs.

Reviewer: Instead of this, I'd include some bullet points with a hyperlink to every command

- [Command A](https://github.com)
- [Command B](https://github.com)
- [Command C](https://github.com)

### Removing an Entra ID / Azure AD Group

Using the CLI for Microsoft 365 it's already possible to remove Microsoft 365 groups. We've now added a new command to remove other types of Entra ID Groups.

To remove an Entra ID Group by display name, run:

```sh
m365 aad group remove --displayName Developers
``` 

> To learn more about managing Entra ID Groups, see the [aad group commands](https://pnp.github.io/cli-microsoft365/cmd/aad/group/group-remove) in our docs.

Reviewer: Also here it would be more useful to just list the commands. This means that the reader has to do a lot less effort to look which other commands are available.

Version 7 also brings a lot of internal changes to our codebase. One important part of that is that we've moved to using ECMAScript Modules (ESM) instead of CommonJS modules. This may not seem directly relevant. But ESM is the standard module format for JavaScript, and offers a number of advantages over CommonJS, including better performance, improved debugging capabilities, and better support for tree-shaking and other optimization techniques. All this will improve the general experience of using the CLI for Microsoft 365. 

### What's next

Here are some things that we are currently working on.

#### Advanced integration with PowerShell

The CLI for Microsoft 365 is a cross platform / cross shell tool. This has always been an important feature. However, many of our users are familiar and use PowerShell in their everyday life. To make the end-to-end experience for these users even better, we're working on some interesting ideas to allow the CLI for Microsoft 365 to work even better when used with PowerShell.

We're currently [working this out](https://github.com/pnp/cli-microsoft365/issues/5470). We'd appreciate your feedback if this is of interest to your area.

> #### Even more commands...

Reviewer 2: no sure. This chapter seems a bit incomplete 😅. I would either remove it or add a bit more light and info and maybe links to issues or developed commands

We're also implementing even more commands to work with Entra ID / Azure AD Groups, the tenant recycle bin, files and loads of further changes and enhancements.

## Contributors

This release wouldn't be possible without the help of (in alphabetical order):

- [Adam Wójcik](https://github.com/Adam-it)

> - [[János Dojcsák](https://github.com/dojcsakj)](https://github.com/dojcsakj)

Reviewer 2: ```suggestion
- [János Dojcsák](https://github.com/dojcsakj)](https://github.com/dojcsakj)
```

is this second [ in `[[János...` needed?

> - [János Dojcsák](https://github.com/dojcsakj)](https://github.com/dojcsakj)

Reviewer: ```suggestion
- [János Dojcsák](https://github.com/dojcsakj)
```

### New contributors

A warm welcome to this month's new contributors! (in alphabetical order):

- [Michał Kornet](https://github.com/mkm17)

Thank you all for the time you chose to spend on CLI for Microsoft 365 and for your help to advance it!

### High fives

CLI for Microsoft 365 wouldn't be where it is today if it weren't for our users who provide us with feedback. High fives to the following people who took the time to share their feedback and ideas for improvement with us (in alphabetical order):

- [kanandb](https://github.com/kanandb)
- [mmcveigh80](https://github.com/mmcveigh80)
- [tohsumi-bebio](https://github.com/tohsumi-bebio)
- [Kunj Balkrishna Sangani](https://github.com/kunj-sangani)

### Discussions

CLI for Microsoft 365 is constantly growing and evolving. We're always looking for ways to improve. There are still many areas not covered by the CLI as well as many amazing ideas for unique features the tool might provide. Your feedback will help us make the right decisions in which areas we should focus on. If you want to help out please do join the currently open [discussions](https://github.com/pnp/cli-microsoft365/discussions)

> What else could we simplify? Let us know what you think by helping out with one of our [open issues](https://github.com/pnp/cli-microsoft365/issues) or chime in on our [discussions](https://github.com/pnp/cli-microsoft365/discussions)!

Reviewer: Maybe we should reference Discord instead of GH discussions?

Author: I've reversed the order...

---
title: CLI for Microsoft 365 v7

> date: 2023-09-28T06:00:00.000Z

Reviewer 3: Shouldn't this be the date when we officially release `v7`?

---
title: CLI for Microsoft 365 v7

> date: 2023-09-30T06:00:00.000Z

Reviewer 3: Let's not forget to update the date 😉

We've currently working out ideas on this. If you'd like to pitch in on how this should be built, do come in and [give us some feedback](https://github.com/pnp/cli-microsoft365/issues/3587).

## Contributors

This release wouldn't be possible without the help of (in alphabetical order):

- [Adam Wójcik](https://github.com/Adam-it)
- [János Dojcsák](https://github.com/dojcsakj)
- [Ganesh Sanap](https://github.com/ganesh-sanap)
- [Jasey Waegebaert](https://github.com/Jwaegebaert)
- [Martin Lingstuyl](https://github.com/martinlingstuyl)
- [Mathijs Verbeeck](https://github.com/MathijsVerbeeck)
- [Michał Kornet](https://github.com/mkm17)
- [Michael Milton](https://github.com/multimeric)
- [Milan Holemans](https://github.com/milanholemans)
- [Nanddeep Nachan](https://github.com/nanddeepn)
- [Nico De Cleyre](https://github.com/nicodecleyre)
- [Saurabh Tripathi](https://github.com/Saurabh7019)
- [Smita Nachan](https://github.com/SmitaNachan)
- [Waldek Mastykarz](https://github.com/waldekmastykarz)

### New contributors

> A warm welcome to this month's new contributors! (in alphabetical order):

Reviewer 3: ```suggestion
A warm welcome to this month's new contributor!:
```
